packages/components/popover/src/Popover.test.tsx

```diff
@@ -1,27 +1,318 @@
+/* eslint-disable max-nested-callbacks */
+/* eslint-disable max-lines-per-function */
 import { render, screen } from '@testing-library/react'
 import userEvent from '@testing-library/user-event'
+import { mockResizeObserver } from 'jsdom-testing-mocks'
+import { useState } from 'react'
 import { describe, expect, it, vi } from 'vitest'
 
 import { Popover } from './Popover'
 
+const getSparkComponent = (componentName: string): Element | null => {
+  return document.querySelector(`[data-spark-component="${componentName}"]`)
+}
```

Comment:
IMO for tests I prefer to repeat the code so I don't need to waste time looking for functions